src/foam/nanos/crunch/ruler/SaveUCJDataOnGranted.js

```diff
@@ -44,6 +49,27 @@ foam.CLASS({
               return;
             }
 
+            //UCJ Edit permission check
+            if ( ! authService.check(x, "usercapabilityjunction.update.*") ) {
+              if (oldUcj.getStatus() == CapabilityJunctionStatus.GRANTED && ucj.getStatus() == CapabilityJunctionStatus.GRANTED ) {
+
+                if ( ucj.getSkipEditBehaviour() == true ) {
+                  ucj.setSkipEditBehaviour(false);
+                  return;
+                }
+    
+                // If edit behaviour does nothing we will keep old data
+                var newData = ucj.getData();
+                ucj.setData(oldUcj.getData());
+    
+                var capability = (Capability) ucj.findTargetId(systemX);
+                var editBehaviour = capability.getEditBehaviour();
+                var editor = (Subject) x.get("subject");
+    
+                editBehaviour.maybeApplyEdit(x, systemX, editor, ucj, newData);
+              }
+            }
+
```

Comment:
at around line 79, the test for n isinrenewable that previously was in the rule predicate needs to be checked. 
so that we only update on:
or
- old not-granted -> new granted
- old granted, old not renewable, new renewable
- old granted, new edit, new granted
so a top level test of n granted and old.data != new.data could take care of a bit of it.